FRAuth/FRAuth/Callbacks/DeviceBindingCallback.swift

```diff
@@ -0,0 +1,221 @@
+                promptDescription = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.timeout, let outputValue = output[CBConstants.value] as? Int {
+                timeout = outputValue
+            }
+        }
+        
+        guard userId != nil, userName != nil, challenge != nil, deviceBindingAuthenticationType != nil, title != nil, subtitle != nil, promptDescription != nil else {
+            throw AuthError.invalidCallbackResponse("Missing an output value")
+        }
+        
+        for input in inputs {
+            if let name = input[CBConstants.name] as? String, name.contains(CBConstants.jws) {
+                self.jwsKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.deviceName) {
+                self.deviceNameKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.deviceId) {
+                self.deviceIdKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.clientError) {
+                self.clientErrorKey = name
+            }
+        }
+        
+        guard jwsKey != nil, deviceNameKey != nil, deviceIdKey != nil, clientErrorKey != nil else {
+            throw AuthError.invalidCallbackResponse("Missing an input value")
+        }
+        
+        try super.init(json: json)
+        
+        self.type = callbackType
+        self.response = json
+    }
+    
+    
+    /// Bind the device.
+    /// - Parameter completion Completion block for Device binding result callback
+    func bind(completion: @escaping DeviceBindingResultCallback) {
+        execute(authInterface: nil, deviceId: nil, completion)
+    }
+    
+    
+    /// Helper method to execute binding , signing, show biometric prompt.
+    /// - Parameter authInterface: Interface to find the Authentication Type - procide nil to default to getDeviceBindingAuthenticator()
+    /// - Parameter deviceId: Interface to find the Authentication Type - provide nil to default to FRDevice.currentDevice?.identifier.getIdentifier()
+    /// - Parameter completion: completion Completion block for Device binding result callback
+    open func execute(authInterface: DeviceAuthenticator?,
+                      deviceId: String?,
+                      _ completion: @escaping DeviceBindingResultCallback) {
+        
+        let newAuthInterface = authInterface ?? getDeviceBindingAuthenticator()
+        let newDeviceId = deviceId ?? FRDevice.currentDevice?.identifier.getIdentifier()
+        
+        guard newAuthInterface.isSupported() else {
+            handleException(status: .unsupported(errorMessage: nil), completion: completion)
+            return
+        }
+        
+        newAuthInterface.authenticate(timeout: timeout ?? 60) { result in
```

Comment:
Are we capturing self on this call? might be work changing that to 
"newAuthInterface.authenticate(timeout: timeout ?? 60) {  [unowned self] result in"

And test if we are OK